src/zenml/cli/base.py

```diff
@@ -64,18 +67,31 @@ def init(path: Optional[Path]) -> None:
 
 
 @cli.command("clean")
-@click.option("--yes", "-y", type=click.BOOL, default=False)
+@click.option("--yes", "-y", is_flag=True, default=False)
 def clean(yes: bool = False) -> None:
-    """Clean everything in repository.
+    """Delete all ZenML metadata and artifacts.
+
+    This is a destructive operation, primarily intended for use in development.
 
     Args:
       yes: bool:  (Default value = False)
     """
     if not yes:
-        _ = confirmation(
-            "This will completely delete all pipelines, their associated "
+        confirm = confirmation(
+            "DANGER: This will completely delete all pipelines, their associated "
             "artifacts and metadata ever created in this ZenML repository. "
             "Are you sure you want to proceed?"
         )
 
-    error("Not implemented for this version")
+    if yes or confirm:
+        local_zen_repo_config = Path.cwd() / REPOSITORY_DIRECTORY_NAME
+        global_zen_config = get_global_config_directory()
+        if exists(str(local_zen_repo_config)):
+            rmtree(str(local_zen_repo_config))
+            declare(f"Deleted local ZenML config from {local_zen_repo_config}.")
+        if exists(global_zen_config):
```

Comment:
I'm wondering if we should try to keep the user id as well as the analytics opt-in status when deleting the global config directory. Personally whenever I'm looking to clean out my local setup, what I'm actually trying to do is delete local stores (=artifact/metadata stores) as well as some of the profiles. But that would require far more flexibility from this function, such as specifying profiles that are ignored from deletion.
As long as @htahir1 is fine with deleting the entire config I'm also okay with merging this as-is to speed up your workflow